File: sorting/selectionSort/selectionSort.py
#  O(n^2) time, O(1) space


def selectionSort(array):

    for i in range(len(array)):
        smallestIndex = i
        for j in range(i, len(array)):
            if array[j] < array[smallestIndex]:
                smallestIndex = j

        if smallestIndex != i:
            swap(i, smallestIndex, array)
    return array


def swap(i, j, array):
    array[i], array[j] = array[j], array[i]


# Example usage:
if __name__ == "__main__":
    arr = [8, 5, 2, 9, 5, 6, 3]
    print("Original array:", arr)
    sorted_arr = selectionSort(arr)
    print("Sorted array:", sorted_arr)

File: sorting/selectionSort/selectionSort.js
// O(n^2) time, O(1) space

function swap(i, j, array) {
  [array[i], array[j]] = [array[j], array[i]]
}

function selectionSort(array) {
  for(let i = 0; i < array.length; i++) {
    let smallestIndex = i
    for(let j = i + 1; j < array.length; j++) {
      if(array[j] < array[smallestIndex]) {
        smallestIndex = j
      }
    }
    if(smallestIndex !== i) {
      swap(i, smallestIndex, array)
    }
  }
  return array
}

// Example usage:
const arr = [64, 25, 12, 22, 11];
console.log(selectionSort(arr)); // Output: [11, 12, 22, 25, 64]
